lib/drivers/insuletDriver.js

```diff
@@ -637,6 +696,266 @@ module.exports = function (config) {
+              alarmText: getNameForValue(ALARM_TYPES, alarmValue),
+              explanation: 'PDM error',
+              stopsDelivery: 'unknown'
+            })
+            .done();
+          break;
+        case 39:
+          postsuspend = makeSuspended(alarm);
+          postbasal = makeSuspendBasal(alarm);
+          postalarm = postalarm.with_alarmType('auto_off')
+            .with_payload({
+              alarmText: getNameForValue(ALARM_TYPES, alarmValue),
+              explanation: 'auto-off',
+              stopsDelivery: true
+            })
+            .with_status(postsuspend)
+            .done();
+          break;
+        // History - REMOTE HAZ
+        case 13:
+          postalarm = postalarm.with_alarmType('other')
+            .with_payload({
+              alarmText: getNameForValue(ALARM_TYPES, alarmValue),
+              explanation: 'clock reset alarm',
+              stopsDelivery: false
+            })
+            .done();
+          break;
+        // Pump advisory and hazard alarm (non-History)
+        case 14:
+          postsuspend = makeSuspended(alarm);
+          postbasal = makeSuspendBasal(alarm);
+          postalarm = postalarm.with_alarmType('no_insulin')
+            .with_payload({
+              alarmText: getNameForValue(ALARM_TYPES, alarmValue),
+              explanation: 'empty reservoir',
+              stopsDelivery: true
+            })
+            .with_status(postsuspend)
+            .done();
+          break;
+        case 15:
+        // TODO: clarify with Insulet or get data to figure out whether this is
+        // a warning or the actual auto-off; the spec is confused
+          postsuspend = makeSuspended(alarm);
+          postbasal = makeSuspendBasal(alarm);
+          postalarm = postalarm.with_alarmType('other')
+            .with_payload({
+              alarmText: getNameForValue(ALARM_TYPES, alarmValue),
+              explanation: 'auto-off',
+              stopsDelivery: true
+            })
+            .with_status(postsuspend)
+            .done();
+          break;
+        case 16:
+          postsuspend = makeSuspended(alarm);
+          postbasal = makeSuspendBasal(alarm);
+          postalarm = postalarm.with_alarmType('no_delivery')
+            .with_payload({
+              alarmText: getNameForValue(ALARM_TYPES, alarmValue),
+              explanation: 'pod expired',
+              stopsDelivery: true
+            })
+            .with_status(postsuspend)
+            .done();
+          break;
+        case 17:
+          postsuspend = makeSuspended(alarm);
+          postbasal = makeSuspendBasal(alarm);
+          postalarm = postalarm.with_alarmType('occlusion')
+            .with_payload({
+              alarmText: getNameForValue(ALARM_TYPES, alarmValue),
+              explanation: 'pump site occluded',
+              stopsDelivery: true
+            })
+            .with_status(postsuspend)
+            .done();
+          break;
+        case 18:
```

Comment:
Yeah, I cleaned up the `buildAlarmRecords` with this suggestion and the above - much better, thanks :)